src/packet.rs

```diff
@@ -451,7 +451,21 @@ packets! {
             // 0x3c => UpdateScore { score_name: String, action: ScoreAction }
             0x3d => DisplayScoreboard { position: i8, score_name: String }
             // 0x3e => UpdateTeam { team_name: String, action: TeamAction }
-            // 0x3f => PluginMessage { channel: String, data: Vec<u8>; encode { ... }; decode { ... }; } // PROBLEM: length of `data` comes from packet length
+            0x3f => PluginMessage { channel: String, data: Vec<u8>;
+                encode {
+                    try!(<String as Protocol>::proto_encode(self.channel, dst));
+                    for elt in self.data.iter() {
+                        try!(<u8 as Protocol>::proto_encode(elt, dst));
```

Comment:
Can we not use [`write_all`](http://doc.rust-lang.org/std/io/trait.Write.html#method.write_all) for this instead of a loop?